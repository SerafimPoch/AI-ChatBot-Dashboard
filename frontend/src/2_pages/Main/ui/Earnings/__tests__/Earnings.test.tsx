import React from "react";
import { render } from "@testing-library/react";
import {
  EarningsContainer,
  EarningsTitleWrapper,
  EarningsTitle,
  EarningsContent,
  EarningsItemWrapper,
  EarningsContentHeadlines,
  EarningsContentBoldText,
  EarningsContentLifeWrapper,
  EarningsContentLightText,
  EarningsContentCapacity,
  EarningsChatBotContainer,
  EarningsChatBotName,
  EarningsChatBotModel,
  EarningsProgressBarWrapper,
  EarningsPercentIndicatorWrapper,
  EarningsTimeSwitcherWrapper,
} from "../styles";

describe("Earnings page components", () => {
  it("EarningsContainer renders correctly", () => {
    const { asFragment } = render(<EarningsContainer />);
    expect(asFragment()).toMatchSnapshot();
  });
  it("EarningsTitleWrapper renders correctly", () => {
    const { asFragment } = render(<EarningsTitleWrapper />);
    expect(asFragment()).toMatchSnapshot();
  });
  it("EarningsTitle renders correctly", () => {
    const { asFragment } = render(<EarningsTitle />);
    expect(asFragment()).toMatchSnapshot();
  });
  it("EarningsContent renders correctly", () => {
    const { asFragment } = render(<EarningsContent />);
    expect(asFragment()).toMatchSnapshot();
  });
  it("EarningsItemWrapper renders correctly", () => {
    const { asFragment } = render(<EarningsItemWrapper />);
    expect(asFragment()).toMatchSnapshot();
  });
  it("EarningsContentHeadlines renders correctly", () => {
    const { asFragment } = render(<EarningsContentHeadlines />);
    expect(asFragment()).toMatchSnapshot();
  });
  it("EarningsContentBoldText renders correctly", () => {
    const { asFragment } = render(<EarningsContentBoldText />);
    expect(asFragment()).toMatchSnapshot();
  });
  it("EarningsContentLifeWrapper renders correctly", () => {
    const { asFragment } = render(<EarningsContentLifeWrapper />);
    expect(asFragment()).toMatchSnapshot();
  });
  it("EarningsContentLightText renders correctly", () => {
    const { asFragment } = render(<EarningsContentLightText />);
    expect(asFragment()).toMatchSnapshot();
  });
  it("EarningsContentCapacity renders correctly", () => {
    const { asFragment } = render(<EarningsContentCapacity />);
    expect(asFragment()).toMatchSnapshot();
  });
  it("EarningsChatBotContainer renders correctly", () => {
    const { asFragment } = render(<EarningsChatBotContainer />);
    expect(asFragment()).toMatchSnapshot();
  });

  it("EarningsChatBotName renders correctly", () => {
    const { asFragment } = render(<EarningsChatBotName />);
    expect(asFragment()).toMatchSnapshot();
  });
  it("EarningsChatBotModel renders correctly", () => {
    const { asFragment } = render(<EarningsChatBotModel />);
    expect(asFragment()).toMatchSnapshot();
  });
  it("EarningsProgressBarWrapper renders correctly", () => {
    const { asFragment } = render(<EarningsProgressBarWrapper />);
    expect(asFragment()).toMatchSnapshot();
  });
  it("EarningsPercentIndicatorWrapper renders correctly", () => {
    const { asFragment } = render(<EarningsPercentIndicatorWrapper />);
    expect(asFragment()).toMatchSnapshot();
  });
  it("EarningsTimeSwitcherWrapper renders correctly", () => {
    const { asFragment } = render(<EarningsTimeSwitcherWrapper />);
    expect(asFragment()).toMatchSnapshot();
  });
});
