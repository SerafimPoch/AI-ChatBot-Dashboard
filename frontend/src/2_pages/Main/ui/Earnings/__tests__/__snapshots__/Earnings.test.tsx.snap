// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Earnings page components EarningsChatBotContainer renders correctly 1`] = `
<DocumentFragment>
  .emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  gap: 32px;
}

<div
    class="emotion-0"
  />
</DocumentFragment>
`;

exports[`Earnings page components EarningsChatBotModel renders correctly 1`] = `
<DocumentFragment>
  .emotion-0 {
  font-family: var(--font-sf-pro-rounded);
  font-size: 14px;
  font-weight: 600;
  line-height: 20px;
  color: var(--text-secondary);
}

<p
    class="emotion-0"
  />
</DocumentFragment>
`;

exports[`Earnings page components EarningsChatBotName renders correctly 1`] = `
<DocumentFragment>
  .emotion-0 {
  font-family: var(--font-sf-pro-rounded);
  font-size: 18px;
  font-weight: 500;
  line-height: 24px;
  color: var(--text-primary);
}

<p
    class="emotion-0"
  />
</DocumentFragment>
`;

exports[`Earnings page components EarningsContainer renders correctly 1`] = `
<DocumentFragment>
  .emotion-0 {
  border-radius: 32px;
  border: 2px solid #f2f2f2;
  padding: 24px 32px;
  grid-row: 3;
  grid-column: 1/3;
}

@media (max-width: 1550px) {
  .emotion-0 {
    grid-row: 3;
  }
}

@media (max-width: 1240px) {
  .emotion-0 {
    grid-row: 4;
    grid-column: 1/3;
  }
}

@media (max-width: 1000px) {
  .emotion-0 {
    padding: 24px;
  }
}

<div
    class="emotion-0"
  />
</DocumentFragment>
`;

exports[`Earnings page components EarningsContent renders correctly 1`] = `
<DocumentFragment>
  .emotion-0 {
  display: grid;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  grid-template-columns: 2fr repeat(4, 1fr);
  margin-bottom: 20px;
  border-bottom: 1px solid #e0e0e0;
  padding-bottom: 20px;
}

@media (max-width: 1000px) {
  .emotion-0 {
    grid-template-columns: 1fr;
  }

  .emotion-0>*:not(:first-of-type) {
    display: none;
  }
}

<div
    class="emotion-0"
  />
</DocumentFragment>
`;

exports[`Earnings page components EarningsContentBoldText renders correctly 1`] = `
<DocumentFragment>
  .emotion-0 {
  font-family: var(--font-sf-pro-rounded);
  color: undefined;
  font-size: 20px;
  font-weight: 600;
  line-height: 24px;
  font-size: 16px;
}

<p
    class="emotion-0"
  />
</DocumentFragment>
`;

exports[`Earnings page components EarningsContentCapacity renders correctly 1`] = `
<DocumentFragment>
  .emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: end;
  -ms-flex-pack: end;
  -webkit-justify-content: flex-end;
  justify-content: flex-end;
}

<div
    class="emotion-0"
  />
</DocumentFragment>
`;

exports[`Earnings page components EarningsContentHeadlines renders correctly 1`] = `
<DocumentFragment>
  .emotion-0 {
  font-family: var(--font-sf-pro-rounded);
  font-size: 14px;
  font-weight: 500;
  line-height: 20px;
  color: undefined;
}

<p
    class="emotion-0"
  />
</DocumentFragment>
`;

exports[`Earnings page components EarningsContentLifeWrapper renders correctly 1`] = `
<DocumentFragment>
  .emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  gap: 10px;
}

<div
    class="emotion-0"
  />
</DocumentFragment>
`;

exports[`Earnings page components EarningsContentLightText renders correctly 1`] = `
<DocumentFragment>
  .emotion-0 {
  font-family: var(--font-sf-pro-rounded);
  color: undefined;
  font-size: 20px;
  font-weight: 600;
  line-height: 24px;
  font-size: 16px;
  color: undefined;
}

<p
    class="emotion-0"
  />
</DocumentFragment>
`;

exports[`Earnings page components EarningsItemWrapper renders correctly 1`] = `
<DocumentFragment>
  .emotion-0 {
  display: grid;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  grid-template-columns: 2fr repeat(4, 1fr);
  margin-bottom: 20px;
  border-bottom: 1px solid #e0e0e0;
  padding-bottom: 20px;
}

@media (max-width: 1000px) {
  .emotion-0 {
    grid-template-columns: 1fr;
  }

  .emotion-0>*:not(:first-of-type) {
    display: none;
  }
}

.emotion-0:last-of-type {
  border-bottom: none;
  padding-bottom: 0;
  margin-bottom: 0;
}

<div
    class="emotion-0"
  />
</DocumentFragment>
`;

exports[`Earnings page components EarningsPercentIndicatorWrapper renders correctly 1`] = `
<DocumentFragment>
  @media (max-width: 1000px) {
  .emotion-0 {
    display: none;
  }
}

<div
    class="emotion-0"
  />
</DocumentFragment>
`;

exports[`Earnings page components EarningsProgressBarWrapper renders correctly 1`] = `
<DocumentFragment>
  @media (max-width: 1000px) {
  .emotion-0 {
    display: none;
  }
}

<div
    class="emotion-0"
  />
</DocumentFragment>
`;

exports[`Earnings page components EarningsTimeSwitcherWrapper renders correctly 1`] = `
<DocumentFragment>
  @media (max-width: 1000px) {
  .emotion-0 {
    display: none;
  }
}

<div
    class="emotion-0"
  />
</DocumentFragment>
`;

exports[`Earnings page components EarningsTitle renders correctly 1`] = `
<DocumentFragment>
  .emotion-0 {
  font-family: var(--font-sf-pro-rounded);
  color: undefined;
  font-size: 20px;
  font-weight: 600;
  line-height: 24px;
}

<p
    class="emotion-0"
  />
</DocumentFragment>
`;

exports[`Earnings page components EarningsTitleWrapper renders correctly 1`] = `
<DocumentFragment>
  .emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  justify-content: space-between;
  margin-bottom: 24px;
}

<div
    class="emotion-0"
  />
</DocumentFragment>
`;
